src/rpc/util.cpp

```diff
@@ -66,9 +66,7 @@ void RPCTypeCheckObj(const UniValue& o,
             throw JSONRPCError(RPC_TYPE_ERROR, strprintf("Missing %s", t.first));
 
         if (!(t.second.typeAny || v.type() == t.second.type || (fAllowNull && v.isNull()))) {
-            std::string err = strprintf("Expected type %s for %s, got %s",
-                uvTypeName(t.second.type), t.first, uvTypeName(v.type()));
-            throw JSONRPCError(RPC_TYPE_ERROR, err);
+            RPCTypeCheckArgument(v, t.second);
```

Comment:
The check is duplicated in this way:

First statement
```
if (!(t.second.typeAny || v.type() == t.second.type || (fAllowNull && v.isNull()))) {
```

Then you call `RPCTypeCheckArgument` which contains another:
```
if (!typeExpected.typeAny && value.type() != typeExpected.type) {
```